scipy/_lib/_array_api.py

```diff
@@ -216,3 +219,37 @@ def assert_less(actual, desired, err_msg='', verbose=True, xp=None):
             desired = desired.cpu()
     return np.testing.assert_array_less(actual, desired,
                                         err_msg=err_msg, verbose=verbose)
+
+
+def cov(x, *, xp=None):
+    if xp is None:
+        xp = array_namespace(x)
+
+    X = copy(x, xp=xp)
+    dtype = xp.result_type(X, xp.float64)
+
+    X = atleast_nd(X, ndim=2, xp=xp)
+    X = xp.asarray(X, dtype=dtype)
+
+    avg = xp.mean(X, axis=1)
+    w_sum = xp.asarray(size(X) / size(avg), dtype=avg.dtype)
+    if w_sum.shape != avg.shape:
+        w_sum = copy(xp.broadcast_to(w_sum, avg.shape), xp=xp)
+
+    w_sum = w_sum[0]
```

Comment:
While working on gh-19186 I noticed that this isn't used. Is it needed?

Reply:
No idea, I tried to stay as true to `np.cov` as possible.

Reply:
OK. Since `w_sum` doesn't appear for the rest of the function, the calculation doesn't seem to be needed anymore. I proposed removing it in lucascolley/scipy#7.